environment.go

```diff
@@ -338,8 +338,8 @@ func (e *envState) createGenerateTasksQueue(ctx context.Context) error {
 	}
 	e.remoteQueueGroup = remoteQueueGroup
 
-	e.closers["single-queue-generate-tasks"] = func(ctx context.Context) error {
-		closerCtx := context.Background()
+	e.closers["generate-tasks"] = func(ctx context.Context) error {
```

Comment:
use the context from the function signature rather than making your own

Reply:
done